.github/workflows/scanbuild_static-analysis.yml

```diff
@@ -72,6 +72,7 @@ jobs:
         apt-get update
         apt-get install -y libffi-dev unzip python libxml2-dev libfindbin-libs-perl make gcc g++ git jq moreutils
     - uses: actions/checkout@v2
+      # needed?
       with:
         fetch-depth: 100
```

Comment:
@pmatos

Is `fetch-depth` still needed? In `ci-pr.yml`, for example, it doesn't seem to require `fetch-depth`.

Reply:
well, it was to make the cloning faster. I think it still makes sense, right? Cloning is faster by quite a bit.

Reply:
Well, not indicating `fetch-depth` means it will fetch only one commit, so it should be faster than fetching 100 commits. The question is, is there any reason why we are fetching 100 commits? 

It used to be that no `fetch-depth` means "no limit". However, that's changed since version 2 beta (which was in December 2019): https://github.com/actions/checkout/commit/e347bba93bdcadab0b55e4b333254f9bb40bdb0c#diff-1243c5424efaaa19bd8e813c5e6f6da46316e63761421b3e5f5c8ced9a36e6b6L18